### ISSUE TITLE
Error: object is not a function on svgLoadedCallback

### ISSUE THREAD

**Issue by joemcgill** [filamentgroup/grunticon#230]
Trying to use the new enhanceSVG option to inject SVGs into a page and everything seems to be working except that I'm getting an 'Uncaught TypeError: object is not a function' error that point to this section of the script loader:

```
    "svg" === e.method && o(function() {
      i(c(a(e.href))), t && t()
    })
```

In my `<head>` I've copied over the v2.0.2 stylesheet loader that Grunticon creates, followed by the following:

```
grunticon(["{{path-to-my-icons}}/icons.data.svg.css", "{{path-to-my-icons}}/icons.data.png.css", "{{path-to-my-icons}}/icons.fallback.css"], grunticon.svgLoadedCallback);
```

Any ideas?

**Comment by jefflembeck** [comment]
@joemcgill Two things: 1. Can you try updating to 2.1? 2. Can you, after doing so, make sure your Gruntfile has `enhanceSVG` set to `true` in your options, run grunticon, and then paste up the loader that you're given in here (if that loader doesn't work after that)?

**Comment by joemcgill** [comment]
According to the package.json file in my node-modules folder, I am using version 2.1.

Here's my grunt task:

```
			grunticon: {
				icons: {
					options: {
						cssprefix: '.iph-icon-',
						enhanceSVG: true
					},
					files: [{
						expand: true,
						cwd: '_icons/dist/svg',
						src: ['*.svg'],
						dest: "_icons/dist"
					}],
				}
			},
```

Here's the loader that's output:

```
/*! grunt-grunticon Stylesheet Loader - v2.0.2 | https://github.com/filamentgroup/grunticon | (c) 2015 Scott Jehl, Filament Group, Inc. | MIT license. */

(function(e){function t(t,n,r,o){"use strict";function a(){for(var e,n=0;u.length>n;n++)u[n].href&&u[n].href.indexOf(t)>-1&&(e=!0);e?i.media=r||"all":setTimeout(a)}var i=e.document.createElement("link"),c=n||e.document.getElementsByTagName("script")[0],u=e.document.styleSheets;return i.rel="stylesheet",i.href=t,i.media="only x",i.onload=o||function(){},c.parentNode.insertBefore(i,c),a(),i}var n=function(r,o){"use strict";if(r&&3===r.length){var a=e.navigator,i=e.Image,c=!(!document.createElementNS||!document.createElementNS("http://www.w3.org/2000/svg","svg").createSVGRect||!document.implementation.hasFeature("http://www.w3.org/TR/SVG11/feature#Image","1.1")||e.opera&&-1===a.userAgent.indexOf("Chrome")||-1!==a.userAgent.indexOf("Series40")),u=new i;u.onerror=function(){n.method="png",n.href=r[2],t(r[2])},u.onload=function(){var e=1===u.width&&1===u.height,a=r[e&&c?0:e?1:2];n.method=e&&c?"svg":e?"datapng":"png",n.href=a,t(a,null,null,o)},u.src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///ywAAAAAAQABAAACAUwAOw==",document.documentElement.className+=" grunticon"}};n.loadCSS=t,e.grunticon=n})(this);(function(e,t){"use strict";var n=t.document,r="grunticon:",o=function(e){if(n.attachEvent?"complete"===n.readyState:"loading"!==n.readyState)e();else{var t=!1;n.addEventListener("readystatechange",function(){t||(t=!0,e())},!1)}},a=function(e){return t.document.querySelector('link[href$="'+e+'"]')},c=function(e){var t,n,o,a,c,i,u={};if(t=e.sheet,!t)return u;n=t.cssRules?t.cssRules:t.rules;for(var s=0;n.length>s;s++)o=n[s].cssText,a=r+n[s].selectorText,c=o.split(");")[0].match(/US\-ASCII\,([^"']+)/),c&&c[1]&&(i=decodeURIComponent(c[1]),u[a]=i);return u},i=function(e){var t,o,a;o="data-grunticon-embed";for(var c in e)if(a=c.slice(r.length),t=n.querySelectorAll(a+"["+o+"]"),t.length)for(var i=0;t.length>i;i++)t[i].innerHTML=e[c],t[i].style.backgroundImage="none",t[i].removeAttribute(o);return t},u=function(t){"svg"===e.method&&o(function(){i(c(a(e.href))),t&&t()})};e.embedIcons=i,e.getCSS=a,e.getIcons=c,e.ready=o,e.svgLoadedCallback=u,e.embedSVG=u})(grunticon,this);
```

**Comment by jefflembeck** [comment]
Got it. Fixed in 2.1.1 - Thanks!

**Comment by joemcgill** [comment]
You're a freaking wizard Lembeck. :clap:

**Comment by matthewmcvickar** [comment]
This is happening to me now.

I'm using Grunticon 2.1.2.

Here's my Grunt task:

```coffeescript
grunticon:
  files:
    expand: true
    cwd: 'tmp/icons'
    src: ['*.{svg,png}']
    dest: 'theme/public/icons'
  options:
    enhanceSVG: true
    corsEmbed: true
```

And the loader that is generated:

```
<script>
/*! grunt-grunticon Stylesheet Loader - v2.1.2 | https://github.com/filamentgroup/grunticon | (c) 2015 Scott Jehl, Filament Group, Inc. | MIT license. */

(function(e){function t(t,n,r,o){"use strict";function a(){for(var e,n=0;u.length>n;n++)u[n].href&&u[n].href.indexOf(t)>-1&&(e=!0);e?i.media=r||"all":setTimeout(a)}var i=e.document.createElement("link"),l=n||e.document.getElementsByTagName("script")[0],u=e.document.styleSheets;return i.rel="stylesheet",i.href=t,i.media="only x",i.onload=o||null,l.parentNode.insertBefore(i,l),a(),i}var n=function(r,o){"use strict";if(r&&3===r.length){var a=e.navigator,i=e.Image,l=!(!document.createElementNS||!document.createElementNS("http://www.w3.org/2000/svg","svg").createSVGRect||!document.implementation.hasFeature("http://www.w3.org/TR/SVG11/feature#Image","1.1")||e.opera&&-1===a.userAgent.indexOf("Chrome")||-1!==a.userAgent.indexOf("Series40")),u=new i;u.onerror=function(){n.method="png",n.href=r[2],t(r[2])},u.onload=function(){var e=1===u.width&&1===u.height,a=r[e&&l?0:e?1:2];n.method=e&&l?"svg":e?"datapng":"png",n.href=a,t(a,null,null,o)},u.src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///ywAAAAAAQABAAACAUwAOw==",document.documentElement.className+=" grunticon"}};n.loadCSS=t,e.grunticon=n})(this);(function(e,t){"use strict";var n=t.document,r="grunticon:",o=function(e){if(n.attachEvent?"complete"===n.readyState:"loading"!==n.readyState)e();else{var t=!1;n.addEventListener("readystatechange",function(){t||(t=!0,e())},!1)}},a=function(e){return t.document.querySelector('link[href$="'+e+'"]')},c=function(e){var t,n,o,a,c,i,u={};if(t=e.sheet,!t)return u;n=t.cssRules?t.cssRules:t.rules;for(var l=0;n.length>l;l++)o=n[l].cssText,a=r+n[l].selectorText,c=o.split(");")[0].match(/US\-ASCII\,([^"']+)/),c&&c[1]&&(i=decodeURIComponent(c[1]),u[a]=i);return u},i=function(e){var t,o,a;o="data-grunticon-embed";for(var c in e)if(a=c.slice(r.length),t=n.querySelectorAll(a+"["+o+"]"),t.length)for(var i=0;t.length>i;i++)t[i].innerHTML=e[c],t[i].style.backgroundImage="none",t[i].removeAttribute(o);return t},u=function(t){"svg"===e.method&&o(function(){i(c(a(e.href))),"function"==typeof t&&t()})};e.embedIcons=i,e.getCSS=a,e.getIcons=c,e.ready=o,e.svgLoadedCallback=u,e.embedSVG=u})(grunticon,this);(function(e,t){"use strict";var n=t.document,r=function(e,n){var r=new t.XMLHttpRequest;return"withCredentials"in r?r.open("GET",e,!0):t.XDomainRequest!==void 0&&(r=new t.XDomainRequest,r.open("GET",e)),n&&(r.onload=n),r.send(),r},o=function(t){"svg"===e.method&&e.ready(function(){r(e.href,function(){var r=n.createElement("style");r.innerHTML=this.responseText;var o=e.getCSS(e.href);o.parentNode.insertBefore(r,o),o.parentNode.removeChild(o),e.embedIcons(e.getIcons(r)),t&&t()})})};e.ajaxGet=r,e.svgLoadedCORSCallback=o,e.embedSVGCors=o})(grunticon,this);

grunticon(["icons.data.svg.css", "icons.data.png.css", "icons.fallback.css"]);
</script>
<noscript><link href="icons.fallback.css" rel="stylesheet"></noscript>
```

And the code I'm using:

```
<script>
(function(e){function t(t,n,r,o){"use strict";function a(){for(var e,n=0;u.length>n;n++)u[n].href&&u[n].href.indexOf(t)>-1&&(e=!0);e?i.media=r||"all":setTimeout(a)}var i=e.document.createElement("link"),l=n||e.document.getElementsByTagName("script")[0],u=e.document.styleSheets;return i.rel="stylesheet",i.href=t,i.media="only x",i.onload=o||null,l.parentNode.insertBefore(i,l),a(),i}var n=function(r,o){"use strict";if(r&&3===r.length){var a=e.navigator,i=e.Image,l=!(!document.createElementNS||!document.createElementNS("http://www.w3.org/2000/svg","svg").createSVGRect||!document.implementation.hasFeature("http://www.w3.org/TR/SVG11/feature#Image","1.1")||e.opera&&-1===a.userAgent.indexOf("Chrome")||-1!==a.userAgent.indexOf("Series40")),u=new i;u.onerror=function(){n.method="png",n.href=r[2],t(r[2])},u.onload=function(){var e=1===u.width&&1===u.height,a=r[e&&l?0:e?1:2];n.method=e&&l?"svg":e?"datapng":"png",n.href=a,t(a,null,null,o)},u.src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///ywAAAAAAQABAAACAUwAOw==",document.documentElement.className+=" grunticon"}};n.loadCSS=t,e.grunticon=n})(this);(function(e,t){"use strict";var n=t.document,r="grunticon:",o=function(e){if(n.attachEvent?"complete"===n.readyState:"loading"!==n.readyState)e();else{var t=!1;n.addEventListener("readystatechange",function(){t||(t=!0,e())},!1)}},a=function(e){return t.document.querySelector('link[href$="'+e+'"]')},c=function(e){var t,n,o,a,c,i,u={};if(t=e.sheet,!t)return u;n=t.cssRules?t.cssRules:t.rules;for(var l=0;n.length>l;l++)o=n[l].cssText,a=r+n[l].selectorText,c=o.split(");")[0].match(/US\-ASCII\,([^"']+)/),c&&c[1]&&(i=decodeURIComponent(c[1]),u[a]=i);return u},i=function(e){var t,o,a;o="data-grunticon-embed";for(var c in e)if(a=c.slice(r.length),t=n.querySelectorAll(a+"["+o+"]"),t.length)for(var i=0;t.length>i;i++)t[i].innerHTML=e[c],t[i].style.backgroundImage="none",t[i].removeAttribute(o);return t},u=function(t){"svg"===e.method&&o(function(){i(c(a(e.href))),"function"==typeof t&&t()})};e.embedIcons=i,e.getCSS=a,e.getIcons=c,e.ready=o,e.svgLoadedCallback=u,e.embedSVG=u})(grunticon,this);(function(e,t){"use strict";var n=t.document,r=function(e,n){var r=new t.XMLHttpRequest;return"withCredentials"in r?r.open("GET",e,!0):t.XDomainRequest!==void 0&&(r=new t.XDomainRequest,r.open("GET",e)),n&&(r.onload=n),r.send(),r},o=function(t){"svg"===e.method&&e.ready(function(){r(e.href,function(){var r=n.createElement("style");r.innerHTML=this.responseText;var o=e.getCSS(e.href);o.parentNode.insertBefore(r,o),o.parentNode.removeChild(o),e.embedIcons(e.getIcons(r)),t&&t()})})};e.ajaxGet=r,e.svgLoadedCORSCallback=o,e.embedSVGCors=o})(grunticon,this);
grunticon(["/theme/public/icons/icons.data.svg.css", "/theme/public/icons/icons.data.png.css", "/theme/public/icons/icons.fallback.css"], grunticon.svgLoadedCORSCallback);
</script>
<noscript><link href="/theme/public/icons/icons.fallback.css" rel="stylesheet"></noscript>
```

Like OP, my SVGs are being injected without incident.

**Comment by jefflembeck** [comment]
Fixed in 2.1.4, thanks @matthewmcvickar

**Comment by matthewmcvickar** [comment]
Perfect! Thank you very much, @jefflembeck.
